### ISSUE TITLE
Frequently no sound

### ISSUE THREAD

**Issue by hepcat72** [popcorntime/popcorntime#2729]
Operating System Version:
macOS Monterey 12.6.4

Popcorn Time Version:
0.4.9

Download date:
couple weeks ago, I think?

Download url:
It was from the releases on this github repo (the link was still purple in the google results when I googled "popcorn-time 0.4.9")

#### Expected Behaviour
Playing anything should have sound

#### Actual Behaviour
About 25% of the time, playing something has no sound and when it happens, every video is like that. If I let it download and open the downloaded file from the cache, it plays with sound in VLC.

#### Steps to reproduce the behaviour
I cannot make it happen.  It seems random, but once it happens, I can't make any movie have sound.

#### Screenshot(s) of issue or error(s) logs of developer console (Windows/Linux: F12, MacOS: ⌘ + 0 ... then 'console' tab) (recommended)
```
error	21:04:10.839615-0400	kernel	1 duplicate report for Sandbox: locationd(143) deny(1) file-read-data /Applications/Popcorn-Time.app
default	21:05:00.125619-0400	runningboardd	Invalidating assertion 184-170-491768 (target:[app<application.com.nw-builder.popcorn-time.165167789.165167799(501)>:48786]) from originator [daemon<com.apple.WindowServer(88)>:170]
default	21:05:00.243220-0400	runningboardd	Calculated state for app<application.com.nw-builder.popcorn-time.165167789.165167799(501)>: running-active (role: UserInteractiveNonFocal)
default	21:05:00.242281-0400	runningboardd	[app<application.com.nw-builder.popcorn-time.165167789.165167799(501)>:48786] Ignoring jetsam update because this process is not memory-managed
default	21:05:00.243927-0400	runningboardd	[app<application.com.nw-builder.popcorn-time.165167789.165167799(501)>:48786] Ignoring suspend because this process is not lifecycle managed
default	21:05:00.243985-0400	runningboardd	[app<application.com.nw-builder.popcorn-time.165167789.165167799(501)>:48786] Ignoring GPU update because this process is not GPU managed
```

**Comment by dutchy1001** [comment]
Have you tried a "clean" (re-)install?
Delete Popcorn and remove the folder in "C:\Users\YOURNAME\AppData\Local\Popcorn-Time.

As you mentioned you can play it after download with VLC, have you tried to watch it direct with VLC . To do this you need to click the little arrow on the "Watch Now" button and choose VLC from the list.

**Comment by kiriles90** [comment]
@hepcat72 I guess you've used the method for macOS to compile with a new version of nwjs instead of the older custom build we use? As a result what's missing from that are many of the audio codecs for the native player. This is also the reason why we don't just update nwjs with a newer vanilla version and instead there is an open issue about it with issues to overcome first. Closing this.

**Comment by hepcat72** [comment]
I'll give it a shot. And I assume the apple equivalent is something like `~/Library/Application Support...`.

I can't remember the issue, but I have tried playing via VLC in the past, and for some reason, it was too annoying, but I'll try that too.

**Comment by hepcat72** [comment]
> @hepcat72 I guess you've used the method for macOS to compile with a new version of nwjs instead of the older custom build we use? As a result what's missing from that are many of the audio codecs for the native player. This is also the reason why we don't just update nwjs with a newer vanilla version and instead there is an open issue about it with issues to overcome first. Closing this.

I don't know anything about that. I used the pkg in the releases. Is that what you mean? What's nwjs?

**Comment by hepcat72** [comment]
I just tried the same video again and found another video that *did* have sound.  I searched the console again and this time I did find references to nwjs.  Will the clean re-install address that?
```
error	07:42:52.460393-0400	tccd	Prompting policy for hardened runtime; service: kTCCServiceAppleEvents requires entitlement com.apple.security.automation.apple-events but it is missing for responsible={<TCCDProcess: identifier=<ID of InvalidCode>, pid=55357, auid=501, euid=501, responsible_path=/Applications/Popcorn-Time.app/Contents/MacOS/nwjs, binary_path=/Applications/Popcorn-Time.app/Contents/MacOS/nwjs>}, accessing={<TCCDProcess: identifier=org.videolan.vlc, pid=55475, auid=501, euid=501, binary_path=/Applications/VLC.app/Contents/MacOS/VLC>}, requesting={<TCCDProcess: identifier=com.apple.appleeventsd, pid=389, auid=55, euid=55, binary_path=/System/Library/CoreServices/appleeventsd>},
error	07:43:27.855833-0400	nwjs Helper (GPU)	 ca_debug_string: conv == NULL
error	07:43:27.860846-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:27.864068-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:27.866312-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:27.868104-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:27.869812-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:27.873235-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:29.873191-0400	tccd	Prompting policy for hardened runtime; service: kTCCServiceAppleEvents requires entitlement com.apple.security.automation.apple-events but it is missing for responsible={<TCCDProcess: identifier=<ID of InvalidCode>, pid=55357, auid=501, euid=501, responsible_path=/Applications/Popcorn-Time.app/Contents/MacOS/nwjs, binary_path=/Applications/Popcorn-Time.app/Contents/MacOS/nwjs>}, accessing={<TCCDProcess: identifier=org.videolan.vlc, pid=55482, auid=501, euid=501, binary_path=/Applications/VLC.app/Contents/MacOS/VLC>}, requesting={<TCCDProcess: identifier=com.apple.appleeventsd, pid=389, auid=55, euid=55, binary_path=/System/Library/CoreServices/appleeventsd>},
error	07:43:40.031615-0400	nwjs Helper (GPU)	 ca_debug_string: conv == NULL
error	07:43:40.057446-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:40.058028-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:40.059559-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:40.063148-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:40.078423-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:40.080491-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:40.081624-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:40.530055-0400	nwjs Helper (GPU)	 ca_debug_string: conv == NULL
error	07:43:40.533402-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:40.533839-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:40.534492-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:40.583236-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:40.634740-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:41.082017-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:51.583149-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:51.586679-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:51.587035-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:51.587542-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
error	07:43:52.128180-0400	mediaremoted	PID Mismatch: Client (null)-55357 is trying to making a nowPlayingClient with a different bundleID com.nw-builder.popcorn-time
```

**Comment by hepcat72** [comment]
It's possible I may have tried a `brew` install or something (which I think, failed) before I found the pkg download. There are so many "popcorn times" out there, each time I need to upgrade to resolve an issue, it's always difficult finding the "same one", i.e. the repo that works for me (this one). And then, the means by which I install - I can never remember what the install process is. I just happened to find my way to the releases a couple weeks ago. Even finding this repo yesterday, I looked for a github link in the app, and couldn't remember how I'd found it before, so I'd googled the app name and the version number from the "about" window.

Anyway, if I try the clean install, I assume I'll lose some settings and/or my history?  Is there anything I can do to preserve that, if that's the case...

**Comment by hepcat72** [comment]
...and using VLC seems to work.  I think that the annoyance I'd had with it in the past had to do with it not being tolerant of buffering issues.  Restarting it had to be done manually, as opposed to in PT, where it restarted on its own.  That was a really old PT version though.

**Comment by kiriles90** [comment]
@hepcat72 just download the macOS .zip file in the releases page here and clean install it. It should resolve the issue and yes you will lose settings and bookmarks. You can preserve them by exporting your database (in the settings page) but then only import them in the new install after you've finished troubleshooting this in case they are the cause.  

Btw to clean install you need to delete  
`~/Library/Application Support/Popcorn-Time`
`~/Library/Cache/Popcorn-Time`
`~/Library/Saved Application State/com.nw-builder.popcorn-time.savedState`
`~/Library/Preferences/com.nw-builder.popcorn-time.plist`  

though.. if VLC works for you and in case you are using the new nwjs method I'd say dont downgrade, it might cause more issues on macOS 12+ (thats why the new nwjs method exists, the old one is unstable on macOS 12 and later, the new one though.. is vanilla nwjs, no extra audio codecs hence the issue)

**Comment by hepcat72** [comment]
Ooooh! Thanks so much.  Incidentally, I noticed I still had the older version of PT, which I'd renamed `Popcorn-Time-old.app`.  I tried running it, out of curiosity.  It gives me an error dialog that just goes away in 2 seconds.  But, when I try playing the same video, it *does* have sound.

![pterr](https://github.com/popcorn-official/popcorn-desktop/assets/2300532/5dfce6fd-9616-415b-8abd-ab85792e2b26)

The newer PT app doesn't give me that error dialog.

**Comment by hepcat72** [comment]
OK, so F12 *in the PT app* and in a browser-like javascript console (I was trying to enter F12 in the system console, lol). And I was searching for that string, which just showed me a line of code.  Came back here to ask for clarification, then I realized you meant to enter it as a *command* in the javascript console. Sigh, I hope it's not going to be one of those days.  Haven't had all my coffee yet, so I'll chalk it up to that. Anyway, the version in the newer app is `0.64.0`.  When I run the older app, it is `0.44.5`.

**Comment by hepcat72** [comment]
OK, cool. I followed that issue. When it closes, I'll update. Until then, I'll use VLC. Unless I run into buffering issues with that again. Thanks for the hand holding.

**Comment by kiriles90** [comment]
@hepcat72 Btw, if you don't like VLC, on macOS I've heard great words about IINA. I'm not a macOS user though and tbh I use VLC as my main video player so I have no personal experience with it.

**Comment by hepcat72** [comment]
I actually love VLC. The issue I encountered when I'd tried to use it in the past, for streaming content from PT, if I'm remembering correctly, was when I was trying to stream a video initiated by PT that wasn't downloading very quickly. Whenever it needed to buffer, it just would stop and I had to manually restart it.

But it's been awhile since I last tried it. If I run into that issue again, I'll see if maybe there exists an update for VLC or something.
